kvbc/src/categorization/kv_blockchain.cpp

```diff
@@ -160,88 +211,105 @@ std::future<BlockDigest> KeyValueBlockchain::computeParentBlockDigest(const Bloc
 
 /////////////////////// Readers ///////////////////////
 
-const Category& KeyValueBlockchain::getCategory(const std::string& cat_id) const {
-  auto it = categorires_.find(cat_id);
-  if (it == categorires_.cend()) {
-    throw std::runtime_error{"Category does not exist = " + cat_id};
+const Category* KeyValueBlockchain::getCategoryPtr(const std::string& cat_id) const {
+  auto it = categories_.find(cat_id);
+  if (it == categories_.cend()) {
+    return nullptr;
   }
-  return it->second;
+  return &it->second;
 }
 
-Category& KeyValueBlockchain::getCategory(const std::string& cat_id) {
-  auto it = categorires_.find(cat_id);
-  if (it == categorires_.end()) {
+Category* KeyValueBlockchain::getCategoryPtr(const std::string& cat_id) {
+  auto it = categories_.find(cat_id);
+  if (it == categories_.end()) {
+    return nullptr;
+  }
+  return &it->second;
+}
+
+const Category& KeyValueBlockchain::getCategoryRef(const std::string& cat_id) const {
+  const auto category = getCategoryPtr(cat_id);
+  if (!category) {
     throw std::runtime_error{"Category does not exist = " + cat_id};
   }
-  return it->second;
+  return *category;
 }
 
-bool KeyValueBlockchain::hasCategory(const std::string& cat_id) const { return (categorires_.count(cat_id) != 0); }
+Category& KeyValueBlockchain::getCategoryRef(const std::string& cat_id) {
+  const auto category = getCategoryPtr(cat_id);
+  if (!category) {
+    throw std::runtime_error{"Category does not exist = " + cat_id};
+  }
+  return *category;
+}
 
 std::optional<Value> KeyValueBlockchain::get(const std::string& category_id,
                                              const std::string& key,
                                              BlockId block_id) const {
-  if (!hasCategory(category_id)) {
+  const auto category = getCategoryPtr(category_id);
+  if (!category) {
     return std::nullopt;
```

Comment:
I know this is maintaining existing behavior, but we may want to consider throwing if a category isn't available. That would be a pretty unexpected error from a user point of view, and it can be confused easily with the key just not existing which is non-exceptional behavior.

Reply:
I think I agree, will put as a TODO. I don't exactly remember why it returns nullopt.